src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -529,4 +528,31 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
             authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
         }
     }
+
+    // TODO: Add doc
+    private void setAuthorRoleOfPostings(List<Post> postsInCourse) {
+        // sets author roles to display user authority icon on posting headers
+        postsInCourse.forEach(post -> {
+            setAuthorRoleForPosting(post, post.getCoursePostBelongsTo());
+            post.getAnswers().forEach(answerPost -> setAuthorRoleForPosting(answerPost, post.getCoursePostBelongsTo()));
+        });
+    }
+
+    // TODO: Add doc
+    private void setAuthorRoleForPosting(Posting posting, Course postingCourse) {
```

Comment:
With the new method you no longer need to pass the course as parameter.
```suggestion
    private void setAuthorRoleForPosting(Posting posting) {
        Course postingCourse = post.getCoursePostBelongsTo();
```

Reply:
I am waiting for some feedback from b-fein. ı thought reviewers would recheck after I click re-request review. From now on I will resolve these conversations after committing. Sorry for the confusion.

Reply:
This method is used jointly for both Posts and AnswerPosts which inherit Postings. The course info is found in Posts but not AnswerPosts. That's why to be able to use the same code snippet and avoid duplication, the course parameter must be passed to the method.

Reply:
Would it make sense to make this an (abstract) method in posting?

Reply:
I don't think so since the function's behavior is the same for both Posts and AnswerPosts. Static maybe, but since the function is in the "PostService" anyways, I think its location is alright.